tests/drivers/flash/testcase.yaml

```diff
@@ -11,3 +11,7 @@ tests:
     platform_allow: nrf52840dk_nrf52840
     tags: nrf52 soc_flash_nrf
     extra_args: OVERLAY_CONFIG=boards/nrf52840_flash_soc.conf
```

Comment:
this line can be removed CONFIG_MPU_ALLOW_FLASH_WRITE is selected if flash driver is enabled since #20314 was merged.

Reply:
nrf52840_flash_soc.conf file is enabling CONFIG_MPU_ALLOW_FLASH_WRITE, so you are suggesting only line#13 to  be removed right?

Reply:
I rather think you can merge `drivers.flash.soc_flash_nrf` and `drivers.flash.stm32` testcases to more generic one: